discard.go

```diff
@@ -21,24 +21,32 @@ import (
 )
 
 // compile time check for compatibility with the telemetry.Logger interface.
-var _ telemetry.Logger = (*discard)(nil)
+var _ telemetry.Logger = (*Discard)(nil)
 
-// discard is a logger that does not emit log messages but still emits metrics.
-type discard struct {
+// Discard is a logger that does not emit log messages but still emits metrics.
+type Discard struct {
 	// ctx holds the Context to extract key-value pairs from to be added to each
 	// log line.
 	ctx context.Context
 	// metric holds the Metric to increment each time Info() or Error() is called.
 	metric telemetry.Metric
 }
 
-// Discard returns a nre logger that discards log messages but still emits the corresponding metrics, if set.
-func Discard() telemetry.Logger {
-	return &discard{ctx: context.Background()}
+// NewDiscard creates a new Discard logger
```

Comment:
Somehow we agreed with @basvanbeek some time ago that ending the godocs with a `dot` was tastier.
```suggestion
// NewDiscard creates a new Discard logger.
```